crates/bevy_asset/src/assets.rs

```diff
@@ -332,6 +341,29 @@ impl AddAsset for App {
             .add_event::<AssetEvent<T>>()
     }
 
+    fn register_asset_reflect<T>(&mut self) -> &mut Self
+    where
+        T: Asset + Reflect + FromReflect + GetTypeRegistration,
+    {
+        let type_registry = self.world.resource::<AppTypeRegistry>();
+        {
+            let mut type_registry = type_registry.write();
+
+            type_registry.register::<T>();
+            type_registry.register::<Handle<T>>();
+            type_registry
+                .get_mut(std::any::TypeId::of::<T>())
+                .unwrap()
+                .insert(<ReflectAsset as FromType<T>>::from_type());
+            type_registry
+                .get_mut(std::any::TypeId::of::<Handle<T>>())
+                .unwrap()
+                .insert(<ReflectHandle as FromType<Handle<T>>>::from_type());
```

Comment:
Nit: Why not use `register_type_data`?
```suggestion
            type_registry.register_type_data::<T, ReflectAsset>();
            type_registry.register_type_data::<Handle<T>, ReflectHandle>();
```